build-support/bin/generate_travis_yml.py

```diff
@@ -248,6 +253,16 @@ def __str__(self) -> str:
         return str(self.value)
 
 
+def _install_rust(homedir: str = "${HOME}") -> List[str]:
+    rustup = (
+        "curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y "
+        "--default-toolchain none"
+    )
+    # This will mutate the PATH to add `rustup` and `cargo`.
+    activate_rustup = f"source {homedir}/.cargo/env"
+    return [rustup, activate_rustup]
```

Comment:
Is it worth doing this in the docker image? Fine as a followup if so.

Reply:
I thought about it, but it would mean duplicating this installation logic across the Dockerfile and still having it in .travis.yml for the non-Docker shards like macOS.

Reply:
I'm fine with us duplicating this installation code if it means that every build avoids downloading rustup. Frankly, all "fixtures" should be installed into the Docker image and never downloaded during a build. Downloading during a build is another way a build could be unstable due to rate limits or general network errors that could (and have likely) occurred.

Reply:
Well, downloading `rustup` might make sense, but actually downloading a toolchain would be a bit much. Would mean you'd have to edit the docker image to adjust `rust-toolchain`.

Reply:
For reference, Toolchain's internal Rust CI image just uses `rustup component add` to add the toolchain into the Docker image.